integration_test/ops_agent_test/main_test.go

```diff
@@ -4491,9 +4632,19 @@ func TestPortsAndAPIHealthChecks(t *testing.T) {
 			t.Fatal(err)
 		}
 
-		cmdOut, err := gce.RunRemotely(ctx, logger, vm, getRecentServiceOutputForImage(vm.ImageSpec))
-		if err != nil {
-			t.Fatal(err)
+		var cmdOut gce.CommandOutput
+		var err error
+		if agents.IsOpsAgentUAPPlugin() {
+			cmdOut, err = gce.RunRemotely(ctx, logger, vm, getHealthCheckLogsForUAPPluginByImage(vm.ImageSpec))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+		} else {
+			cmdOut, err = gce.RunRemotely(ctx, logger, vm, getRecentServiceOutputForImage(vm.ImageSpec))
+			if err != nil {
+				t.Fatal(err)
+			}
```

Comment:
This block seem to be repeated for each of the health check tests. Let's abstract this by introducing a new function `getHealthCheckResultsForImage` that calls one of `getHealthCheckLogsForUAPPluginByImage` or `getRecentServiceOutputForImage`.

Reply:
Done